raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpansProcessor.java

```diff
@@ -169,6 +202,91 @@ public KeyValue<TraceIdentity, StructuredTrace> transform(TraceIdentity key, Raw
     return null;
   }
 
+  private void processMirroringSpan(
+      TraceIdentity key, RawSpan value, TraceState traceState, long currentTimeMs) {
+    Event event = value.getEvent();
+    String tenantId = key.getTenantId();
+    ByteBuffer traceId = value.getTraceId();
+    boolean firstEntry = (traceState == null);
+    final Optional<String> maybeEnvironment =
+        HttpSemanticConventionUtils.getEnvironmentForSpan(event);
+
+    if (SpanSemanticConventionUtils.isClientSpanForOCFormat(
+        event.getAttributes().getAttributeMap())) {
+      final Optional<String> maybeHostAddr = HttpSemanticConventionUtils.getHostIpAddress(event);
+      final Optional<String> maybePeerAddr = HttpSemanticConventionUtils.getPeerIpAddress(event);
+      final Optional<String> maybePeerPort = HttpSemanticConventionUtils.getPeerPort(event);
+      if (maybeEnvironment.isPresent()
```

Comment:
i dont think you will need this complex if checks. you can simply create different IPIdentity and work with them. Also its not necessary that these all will be present. we should work with whatever is provided with some non empty checks

Reply:
environment check we have to make compulsory. For others we can keep optional